sale_stock_medical_prescription_us/models/medical_prescription_order_line.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models
+from datetime import datetime
+
+
+class MedicalPrescriptionOrderLine(models.Model):
+
+    _inherit = 'medical.prescription.order.line'
+
+    refill_qty_remain = fields.Float(
+        string='Refill Remain',
+        store=True,
+        compute='_compute_can_dispense_and_qty',
+        help='Amount of refills remaining in the prescription',
+    )
+    total_qty_remain = fields.Float(
+        string='Qty Remaining',
+        store=True,
+        compute='_compute_can_dispense_and_qty',
+        help='Total units remaining in the prescription',
+    )
+    total_allowed_qty = fields.Float(
+        string='Qty Allowed',
+        store=True,
+        compute='_compute_can_dispense_and_qty',
+        help='Total units allowed in the prescription, including refills',
+    )
+    dispense_remain_qty = fields.Float(
+        string='Dispense Remaining Qty',
+        compute='_compute_dispense_remain',
+        help='Estimated number of units remaining from all finished dispenses',
+    )
+    dispense_remain_day = fields.Float(
+        string='Dispense Remaining Days',
+        compute='_compute_dispense_remain',
+        help='Estimated days remaining based on all finished dispenses',
+    )
+
+    @api.multi
+    def _compute_dispense_remain(self):
+        day_uom_id = self.env.ref('medical_medication.product_uom_day')
+        for record in self:
+
+            if not record.duration_uom_id or record.duration <= 0:
+                continue
+
+            procurements = \
+                record.dispensed_ids.filtered(
+                    lambda r: r.state == 'done'
+                ).sorted(
+                    key=lambda r: r.date_planned
+                )
+
+            if not procurements:
+                record.dispense_remain_qty, record.dispense_remain_day = 0, 0
+                continue
+            first_proc = procurements[0]
+
+            date_first_dispense = fields.Datetime.from_string(
+                first_proc.date_planned,
+            )
+            date_delta = datetime.now() - date_first_dispense
+            days_passed = date_delta.days
+
+            if day_uom_id != record.duration_uom_id:
+
+                # @TODO
```

Comment:
Fix needed: day_uom_id should be 3rd argument and duration_uom_id first arg. Check medication uom data to see if adjusting there can remedy having to put the code the way it is currently.